### PR TITLE
Update WebSocket.h

### PR DESCRIPTION
**Pull request by ismart-omari**
websocket: add a more generic connect function

### CODE REVIEW

File: include/frnetlib/WebSocket.h [hunk 1]
```diff
@@ -42,10 +42,12 @@ namespace fr
          *
          * @param address The address of the socket to connect to
          * @param port The port of the socket to connect to
+         * @param path The path of the resource
+         * @param ws_protocole The Websocket-Protocol, nullptr if NA
          * @param timeout The number of seconds to wait before timing the connection attempt out. Pass {} for default.
          * @return A Socket::Status indicating the status of the operation (Success on success, an error type on failure).
          */
-        Socket::Status connect(const std::string &address, const std::string &port, std::chrono::seconds timeout) override
+        Socket::Status connect(const std::string &address, const std::string &port, const char* path,  const char* ws_protocole, std::chrono::seconds timeout)
```
Comment (Cloaked9000) [inline include/frnetlib/WebSocket.h]: I think `const char* ws_protocole` should be `const std::vector<std::string> &ws_protocols`, for consistencies' sake.

**Comment by Cloaked9000** [comment]
Looks good bar the one comment I've left 🙂

File: include/frnetlib/WebSocket.h [hunk 2]
```diff
@@ -60,8 +60,8 @@ namespace fr
             request.header("sec-websocket-key") = websocket_key;
             request.header("sec-websocket-version") = "13";
 
-            if(ws_protocole)
-                request.header("sec-websocket-protocol") = ws_protocole;
+            if(ws_protocol.size())
```
Comment (Cloaked9000) [inline include/frnetlib/WebSocket.h]:
```suggestion
            if(!ws_protocol.empty())
```

File: include/frnetlib/WebSocket.h (hunk 1 above)
Comment (ismart-omari) [reply include/frnetlib/WebSocket.h]: OK, the vector<> version is done.

**Comment by Cloaked9000** [comment]
Thanks!

**Pull request merged**